src/lbaf/Execution/lbsClusteringTransferStrategy.py

```diff
@@ -99,23 +99,33 @@ def execute(self, known_peers, phase: Phase, ave_load: float):
 
             # Identify and perform beneficial cluster swaps
             n_swaps = 0
-            for o_src in clusters_src.values():
-                swapped_cluster = False
-                for r_try in targets:
+            for r_try in targets if self._deterministic_transfer else random.sample(
+                    targets, len(targets)):
+                # Escape targets loop if at least one swap already occurred
+                if n_swaps:
+                    break
+
+                # Iterate over potential targets to try to swap clusters
+                for k_src, o_src in clusters_src.items():
                     # Iterate over target clusters
-                    for o_try in self.__cluster_objects(r_try).values():
+                    for k_try, o_try in self.__cluster_objects(r_try).items():
                         # Decide whether swap is beneficial
-                        if self._criterion.compute(r_src, o_src, r_try, o_try) > 0.0:
-                            # Perform swap
-                            self._n_transfers += phase.transfer_objects(
-                                r_src, o_src, r_try, o_try)
-                            swapped_cluster = True
-                            n_swaps += 1
-                            break
-
-                    # Break out from targets loop once one swap was performed
-                    if swapped_cluster:
-                        break
+                        c_try = self._criterion.compute(r_src, o_src, r_try, o_try)
+                        if c_try > 0.0:
+                            # Compute source cluster size only when necessary
+                            sz_src = sum([o.get_load() for o in o_src])
+                            if  c_try > 0.05 * sz_src:
```

Comment:
Should this 5% be made into a tunable parameter?

Reply:
Yes. I am creating a follow-on issue for this.

Reply:
@lifflander @cwschilly #434